src/seedu/addressbook/Main.java

```diff
@@ -105,12 +109,15 @@ private void recordResult(CommandResult result) {
      * @param command user command
      * @return result of the command
      */
-    private CommandResult executeCommand(Command command)  {
+    private CommandResult executeCommand(Command command) throws StorageOperationException  {
+        command.setData(addressBook, lastShownList);
+        CommandResult result = command.execute();
         try {
-            command.setData(addressBook, lastShownList);
-            CommandResult result = command.execute();
             storage.save(addressBook);
             return result;
+        } catch (StorageOperationException e) {
+            ui.showToUser("Error writing to Read-Only file");
+            throw new StorageOperationException("Error writing to Read-Only file");
```

Comment:
If you throw the exception and handle it upstream, you'll show the user the message twice. Since it's already handled here, no need to throw it again.